### PR TITLE
S3 event notifications

### PR DESCRIPTION
**Pull request by dminnear**
@Bowbaq check it out when you can!

### CODE REVIEW

File: builtin/providers/aws/resource_aws_s3_bucket.go [hunk 1]
```diff
@@ -740,6 +776,124 @@ func resourceAwsS3BucketWebsiteDelete(s3conn *s3.S3, d *schema.ResourceData) err
 	return nil
 }
 
+func resourceAwsS3BucketNotificationsUpdate(s3conn *s3.S3, d *schema.ResourceData) error {
+	ns := d.Get("notifications").([]interface{})
+
+	if len(ns) == 1 {
```
Comment (Bowbaq) [inline builtin/providers/aws/resource_aws_s3_bucket.go]:
```golang
if len(ns) == 1 {
   notifications := ns[0].(map[string]interface{})
   return resourceAwsS3BucketNotificationsPut(s3conn, d, notifications)
}

if len(ns) == 0 {
  return resourceAwsS3BucketNotificationsDelete(s3conn, d)
}

return fmt.Errorf("Cannot specify more than one set of notitications. Combine json from multiple configurations into a single configuration.")
```

File: builtin/providers/aws/resource_aws_s3_bucket.go [hunk 2]
```diff
@@ -740,6 +776,124 @@ func resourceAwsS3BucketWebsiteDelete(s3conn *s3.S3, d *schema.ResourceData) err
...
+
+func resourceAwsS3BucketNotificationsRead(s3conn *s3.S3, d *schema.ResourceData) error {
+	ns, err := s3conn.GetBucketNotificationConfiguration(
+		&s3.GetBucketNotificationConfigurationRequest{
+			Bucket: aws.String(d.Id()),
+		},
+	)
+
+	var notifications []map[string]interface{}
+	if err == nil {
```
Comment (Bowbaq) [inline builtin/providers/aws/resource_aws_s3_bucket.go]:
```go
if err != nil {
  return err
}
```
then unindent the rest of the block

File: builtin/providers/aws/resource_aws_s3_bucket.go [hunk 3]
```diff
@@ -740,6 +776,124 @@ func resourceAwsS3BucketWebsiteDelete(s3conn *s3.S3, d *schema.ResourceData) err
...
+			}
+			configuration["sqs"] = string(sqs)
+		}
+
+		if v := ns.TopicConfigurations; v != nil {
+			sns, err := json.Marshal(v)
+			if err != nil {
+				return fmt.Errorf("error while marshaling sns notification configuration: %s", err)
+			}
+			log.Printf("[DEBUG] sns string: %s", string(sns))
```
Comment (Bowbaq) [inline builtin/providers/aws/resource_aws_s3_bucket.go]: remove?

**Comment by dminnear** [comment]
@Bowbaq take another look when you can

**Pull request closed without merging**